coderd/insights_test.go

```diff
@@ -570,6 +580,810 @@ func TestTemplateInsights(t *testing.T) {
+						// this user and their app usage will only be counted
+						// once but the template ID will show up in the data.
+						startedAt:          frozenWeekAgo,
+						endedAt:            frozenWeekAgo.Add(time.Hour),
+						sessionCountVSCode: 1,
+						sessionCountSSH:    1,
+					},
+				},
+				appUsage: []appUsage{
+					// TODO(mafredri): This doesn't behave correctly right now
+					// and will add more usage to the app. This could be
+					// considered both correct and incorrect behavior.
+					// { // One hour of usage, but same user and same template app, only count once.
+					// 	app:       users[0].workspaces[1].apps[0],
+					// 	startedAt: frozenWeekAgo,
+					// 	endedAt:   frozenWeekAgo.Add(time.Hour),
+					// 	requests:  1,
+					// },
+					{
+						// Different templates but identical apps, apps will be
+						// combined and usage will be summed.
+						app:       users[0].workspaces[1].apps[0],
+						startedAt: frozenWeekAgo.AddDate(0, 0, 2),
+						endedAt:   frozenWeekAgo.AddDate(0, 0, 2).Add(6 * time.Hour),
+						requests:  1,
+					},
+				},
+			},
+			users[0].workspaces[2]: {
+				agentStats: []agentStat{},
+				appUsage:   []appUsage{},
+			},
+			users[1].workspaces[0]: {
+				agentStats: []agentStat{
+					{ // One hour of usage before timeframe (exclude).
+						startedAt:          frozenWeekAgo.Add(-time.Hour),
+						endedAt:            frozenWeekAgo,
+						sessionCountVSCode: 1,
+						sessionCountSSH:    1,
+					},
+					{ // One hour of usage.
+						startedAt:       frozenWeekAgo,
+						endedAt:         frozenWeekAgo.Add(time.Hour),
+						sessionCountSSH: 1,
+					},
+					{ // One hour of usage after timeframe (exclude in UTC, include in São Paulo).
+						startedAt:          frozenWeekAgo.AddDate(0, 0, 7),
+						endedAt:            frozenWeekAgo.AddDate(0, 0, 7).Add(time.Hour),
+						sessionCountVSCode: 1,
+						sessionCountSSH:    1,
+					},
+				},
+				appUsage: []appUsage{},
+			},
+			users[2].workspaces[0]: {
+				agentStats: []agentStat{
+					{ // One hour of usage.
+						startedAt:                   frozenWeekAgo,
+						endedAt:                     frozenWeekAgo.Add(time.Hour),
+						sessionCountSSH:             1,
+						sessionCountReconnectingPTY: 1,
+					},
+				},
+				appUsage: []appUsage{
+					{
+						app:       users[2].workspaces[0].apps[0],
+						startedAt: frozenWeekAgo.AddDate(0, 0, 2),
+						endedAt:   frozenWeekAgo.AddDate(0, 0, 2).Add(5 * time.Minute),
+						requests:  1,
+					},
+					{ // Excluded from apps, but counted as active during the day.
+						app:       users[2].workspaces[0].apps[1],
+						startedAt: frozenWeekAgo.AddDate(0, 0, 3),
+						endedAt:   frozenWeekAgo.AddDate(0, 0, 3).Add(5 * time.Minute),
+						requests:  1,
+					},
+				},
+			},
+		}
+	}
+	type testRequest struct {
+		name        string
+		makeRequest func([]*testTemplate) codersdk.TemplateInsightsRequest
+		ignoreTimes bool
+	}
+	tests := []struct {
```

Comment:
Do you think that we can convert-to-table/remove some other insights tests?

Reply:
I think we can delete at least one or two. But depending on if we rely primarily on dbgen here or not, we might want to keep some "whole tests" around.

Reply:
I deleted one test `TestTemplateInsights`, its test cases are now covered by the golden files and then some.